### PR TITLE
SLING-10463 - "create path" should fail if a property exists at the same path

### PR DESCRIPTION
**Pull request by bdelacretaz**

### CODE REVIEW

**Comment by anchela** [comment]
@bdelacretaz somehow can't comment directly on the affect lines. here what i would suggest to change.
instead of 
```
        if (session.propertyExists(fullPath)) {
                        throw new RuntimeException("Error creating Node at " + fullPath + ": a property exists with the same path");
                } else if (session.itemExists(fullPath)) {
                        log.info("Path already exists, nothing to do (and not checking its primary type for now");
               } else ...
```
i would rather write:
```
        if (session.propertyExists(fullPath)) {
                        throw new RuntimeException("Error creating Node at " + fullPath + ": a property exists with the same path");
                } else if (session.nodeExists(fullPath)) {
                        log.info("Node already exists, nothing to do (and not checking its primary type for now");
               } else ...
```

File: src/main/java/org/apache/sling/jcr/repoinit/impl/AclVisitor.java [hunk 1]
```diff
@@ -142,7 +142,9 @@ public void visitCreatePath(CreatePath cp) {
         for (PathSegmentDefinition psd : cp.getDefinitions()) {
             final String fullPath = parentPath + "/" + psd.getSegment();
             try {
-                if (session.itemExists(fullPath)) {
+                if (session.propertyExists(fullPath)) {
+                    throw new RuntimeException("Error creating Node at " + fullPath + ": a property exists with the same path");
+                } else if (session.itemExists(fullPath)) {
```
Comment (anchela) [inline src/main/java/org/apache/sling/jcr/repoinit/impl/AclVisitor.java:147]: i would rather use 'session.nodeExists(fullPath)' here just for the sake of clarity and also to make really sure we don't by mistake have a property introduced just now somehow.... not very likely but i have seen all sort of unlikely nonsense.

File: src/main/java/org/apache/sling/jcr/repoinit/impl/AclVisitor.java [hunk 2]
```diff
@@ -142,7 +142,9 @@ public void visitCreatePath(CreatePath cp) {
...
                     log.info("Path already exists, nothing to do (and not checking its primary type for now): {}", fullPath);
```
Comment (anchela) [inline src/main/java/org/apache/sling/jcr/repoinit/impl/AclVisitor.java:148]: log.info("Node at {} already exists.....
that makes it unambiguous that we are trying to create a node and not a property. "Path already exists" is a bit vague....

File: src/test/java/org/apache/sling/jcr/repoinit/CreatePathsTest.java [hunk 3]
```diff
@@ -19,6 +19,7 @@
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.UUID;
```
Comment (anchela) [inline src/test/java/org/apache/sling/jcr/repoinit/CreatePathsTest.java:22]: unused import?

**Review by anchela — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/sling/jcr/repoinit/impl/AclVisitor.java (hunk 2 above)
Comment (bdelacretaz) [reply src/main/java/org/apache/sling/jcr/repoinit/impl/AclVisitor.java]: Agreed, I have made that change and also changed the log message to say "Node"

File: src/test/java/org/apache/sling/jcr/repoinit/CreatePathsTest.java (hunk 3 above)
Comment (bdelacretaz) [reply src/test/java/org/apache/sling/jcr/repoinit/CreatePathsTest.java]: good catch, thanks!

**Comment by bdelacretaz** [comment]
I have updated the code according to the above comments, please review.

**Comment by kwin** [comment]
@bdelacretaz What happened to this PR? To me it looks good and I would appreciate to get this merged after fixing the conflict...

**Comment by kwin** [comment]
Superseded by #41.

**Pull request closed without merging**